source/extensions/path/uri_template_lib/proto/rewrite_segments.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.extensions.uri_template;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.uri_template";
+option java_outer_classname = "UriTemplateRewrite";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/url_template/url_template";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// Holds the segments for rewriting urls base on uri pattern templates
+message RewriteSegments {
```

Comment:
Does this end up getting included in the public API? If so, perhaps it should be tagged `[#not-implemented-hide:]` to hide it. I was going to suggest that perhaps it should be a C++ type, but I see where you use it to convert strings in test cases into objects for testing.

Reply:
This should just be internal. No visibility from public API

Reply:
Hm. If this is internal only, then could it just be a C++ type instead of a proto? 
(+1 to `[#not-implemented-hide:]` if we need to keep it as a proto)

Reply:
I looked at the docs generated in the PR build and this doesn't seem to appear anywhere, so I think it's ok to leave as is.

Reply:
Added todo to convert this to struct